core/org.wso2.carbon.ndatasource.rdbms/src/main/java/org/wso2/carbon/ndatasource/rdbms/CorrelationLogInterceptor.java

```diff
@@ -156,7 +157,15 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
                 Object result = null;
 
                 if (this.delegate != null) {
-                    result = method.invoke(this.delegate, args);
+                    try {
+                        result = method.invoke(this.delegate, args);
+                    } catch (InvocationTargetException e) {
+                        // Since the exception is checked and not declared by the proxied interface, it will first be
+                        // wrapped in an UndeclaredThrowableException. The Method.invoke wraps all exceptions in an
+                        // InvocationTargetException. By re-throwing the InvocationTargetHandler’s cause, the client
+                        // code will be able to see the real cause of the exception.
+                        throw e.getCause();
```

Comment:
Let's do a null check for e.getCause before throwing,
```
                if (e.getCause() != null) {
                    throw e.getCause();
                } else {
                    throw e;
                }
```

Reply:
Fixed https://github.com/wso2/carbon-kernel/pull/3160/commits/78df3676c309f8954c251caddc249154201271bb